apps/mull-ui/src/app/app.tsx

```diff
@@ -68,19 +94,20 @@ export const App = () => {
               />
             }
           />
-          <Route path={ROUTES.HOME}>
+          <PrivateRoute path={ROUTES.HOME}>
             <SubNavBar className="top-nav-bar-shadow" />
             <div className="page-container with-sub-nav-bar">
               <SwipeableRoutes>
-                <Route path={ROUTES.DISCOVER} component={DiscoverPage} />
-                <Route path={ROUTES.UPCOMING} component={UpcomingPage} />
-                <Route path={ROUTES.MY_EVENTS} component={MyEventsPage} />
-                <Route exact path={ROUTES.HOME}>
+                <PrivateRoute path={ROUTES.DISCOVER} component={DiscoverPage} />
+                <PrivateRoute path={ROUTES.UPCOMING} component={UpcomingPage} />
+                <PrivateRoute path={ROUTES.MY_EVENTS} component={MyEventsPage} />
+                <PrivateRoute exact path={ROUTES.HOME}>
                   <Redirect to={ROUTES.DISCOVER} />
-                </Route>
+                </PrivateRoute>
               </SwipeableRoutes>
             </div>
-          </Route>
+          </PrivateRoute>
+          <Route path="*" component={NotFoundPage} />
```

Comment:
We don't need the path prop by default it will match any route that isn't matched
```suggestion
          <Route component={NotFoundPage} />
```